### PR TITLE
remove lockfile if it is empty

### PR DESCRIPTION
**Pull request by mattkasun**

### CODE REVIEW

**Review by afeiszli — APPROVED** [review]

**Comment by afeiszli** [comment]
What issues did this cause?

**Comment by afeiszli** [comment]
Nvm, just saw the ticket. Should we merge this before cutting release?

**Review by abhishek9686 — APPROVED** [review]

File: config/config.go [hunk 1]
```diff
@@ -288,31 +288,26 @@ func Lock(lockfile string) error {
...
 			bytes, err := os.ReadFile(lockfile)
-			if err == nil {
+			if err != nil || len(bytes) == 0 {
+				_ = os.Remove(lockfile)
+			} else {
 				var owner int
-				if json.Unmarshal(bytes, &owner) == nil {
+				if err := json.Unmarshal(bytes, &owner); err != nil {
+					_ = os.Remove(lockfile)
+				} else {
```
Comment (TobiaszCudnik) [inline config/config.go:297]: `else if`

**Review by TobiaszCudnik — APPROVED** [review]

**Review by Aceix — APPROVED** [review]

**Review by theguy951357 — APPROVED** [review]
critical list passes
pr passes

**Pull request merged**